File: Source Codes/Sudoku Solver.py
sudoku=[]
solved= False

def blocked(val,i,j):
    i-=i%3
    j-=j%3

    for x in range(i,i+3):
        for y in range(j,j+3):
            if(sudoku[x][y]==val):
                return True
    return False

def notIllegal(val,i,j):
    for x in range(9):
        if(sudoku[i][x]==val):
            return False

    for x in range(9):
        if(sudoku[x][j]==val):
            return False

    if(blocked(val,i,j)):
        return False

    return True


def solve(cell):

    i= int (cell/9)
    j= int (cell%9)

    if(i==9):
        global solved
        solved= True
        print()
        for x in range(9):
            for y in range(9):
                print(sudoku[x][y], end=" ")
            print()
        return

    if(sudoku[i][j]!=0):
        solve(cell+1)
        return

    for x in range(1,10):
        if(notIllegal(x,i,j)):
            sudoku[i][j]=x
            solve(cell+1)
            if(solved):
                return
            sudoku[i][j]=0 #Back Tacking



if __name__ == '__main__':
    print("/*Place 0 for cells that don't have an entry\nSeperate columns with space and rows with new line*/\n")

    for i in range(0,9):
        row=[]
        inp = list(map(int,input().split()))

        sudoku.append(inp)

    solve(0)

File: Source Codes/Sudoku Solver.cpp
#include<bits/stdc++.h>
using namespace std;

int sudoku[9][9];
bool flag= false;


bool blocked(int val, int i, int j){
    int m,n;

    m= i - (i%3);
    n= j - (j%3);

    for(int x=m; x<m+3;x++){
        for(int y=n;y<n+3;y++){
            if(sudoku[x][y]==val)
                return true;
        }
    }

    return false;


}


bool notIllegal(int val, int i, int j){
    for(int m=0;m<9;m++){
        if(sudoku[m][j]==val)
            return false;
    }

    for(int n=0;n<9;n++){
        if(sudoku[i][n]==val)
            return false;
    }

    if(blocked(val,i,j))
        return false;

    return true;


}


void solve(int cell){
    int i,j;

    i=cell/9;
    j=cell%9;

    if(i==9){
        flag=true;
        puts("");
        for(int m=0;m<9;m++){
            for(int n=0;n<9;n++)
                cout<<sudoku[m][n]<<" ";
            puts("");
        }
        return;
    }

    if(sudoku[i][j]!=0){
        solve(cell+1);
        return;
    }

    for(int x=1; x<=9; x++){
        if(notIllegal(x,i,j)){
            sudoku[i][j]=x;
            solve(cell+1);

            if(flag)
                return;

            sudoku[i][j]=0;
        }

    }



}

main(){

    puts("/*Input 0 For Cells That Don't Have Any Entry*/");

    for(int i=0;i<9;i++)
        for(int j=0;j<9;j++)
            cin>> sudoku[i][j];

    solve(0);




}
